DFC.ServiceTaxonomy.Taxonomies/Controllers/AdminController.cs

```diff
@@ -0,0 +1,317 @@
+            contentItem.Alter<TermPart>(t => t.TaxonomyContentItemId = taxonomyContentItemId);
+
+            dynamic model = await _contentItemDisplayManager.BuildEditorAsync(contentItem, _updateModelAccessor.ModelUpdater, true);
+
+            model.TaxonomyContentItemId = taxonomyContentItemId;
+            model.TaxonomyItemId = taxonomyItemId;
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [ActionName("Create")]
+        public async Task<IActionResult> CreatePost(string id, string taxonomyContentItemId, string taxonomyItemId)
+        {
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageTaxonomies))
+            {
+                return Forbid();
+            }
+
+            ContentItem taxonomy;
+
+            var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition("Taxonomy");
+
+            if (!contentTypeDefinition.GetSettings<ContentTypeSettings>().Draftable)
+            {
+                taxonomy = await _contentManager.GetAsync(taxonomyContentItemId, VersionOptions.Latest);
+            }
+            else
+            {
+                taxonomy = await _contentManager.GetAsync(taxonomyContentItemId, VersionOptions.DraftRequired);
+            }
+
+            if (taxonomy == null)
+            {
+                return NotFound();
+            }
+
+            var contentItem = await _contentManager.NewAsync(id);
+            contentItem.Weld<TermPart>();
+            contentItem.Alter<TermPart>(t => t.TaxonomyContentItemId = taxonomyContentItemId);
+
+            dynamic model = await _contentItemDisplayManager.UpdateEditorAsync(contentItem, _updateModelAccessor.ModelUpdater, true);
+
+            if (!ModelState.IsValid)
+            {
+                model.TaxonomyContentItemId = taxonomyContentItemId;
+                model.TaxonomyItemId = taxonomyItemId;
+
+                return View(model);
+            }
+
+            if (taxonomyItemId == null)
+            {
+                // Use the taxonomy as the parent if no target is specified
+                taxonomy.Alter<TaxonomyPart>(part => part.Terms.Add(contentItem));
+            }
+            else
+            {
+                // Look for the target taxonomy item in the hierarchy
+                var parentTaxonomyItem = FindTaxonomyItem(taxonomy.As<TaxonomyPart>().Content, taxonomyItemId);
+
+                // Couldn't find targeted taxonomy item
+                if (parentTaxonomyItem == null)
+                {
+                    return NotFound();
+                }
+
+                var taxonomyItems = parentTaxonomyItem?.Terms as JArray;
+
+                if (taxonomyItems == null)
+                {
+                    parentTaxonomyItem["Terms"] = taxonomyItems = new JArray();
+                }
+
+                taxonomyItems.Add(JObject.FromObject(contentItem));
+            }
+
+            await _contentManager.PublishAsync(taxonomy);
```

Comment:
Could you please update the Stax startup recipe, so that taxonomies/page locations (as appropriate) can only have published versions (remove Draftable I think).